### PR TITLE
SUS-805: Use user ID in ipblocks when possible

### PR DESCRIPTION
**Pull request by mszabo-wikia**
* `ipb_by` is always a registered user ID - `ipb_by_text` is redundant and can be dropped
* query based on user ID (`ipb_user`) when checking blocked status of registered user

https://wikia-inc.atlassian.net/browse/SUS-805

### CODE REVIEW

File: includes/Block.php [hunk 1]
```diff
@@ -223,8 +229,8 @@ protected function newLoad( $vagueTarget = null ) {
 			list( $target, $type ) = self::parseTarget( $vagueTarget );
 			switch( $type ) {
 				case self::TYPE_USER:
-					# Slightly wierd, but who are we to argue?
```
Comment (macbre) [inline includes/Block.php:226]: :smile:

File: includes/User.php [hunk 2]
```diff
@@ -1350,7 +1350,7 @@ private function getBlockedStatus( $bFromSlave = true, $shouldLogBlockInStats =
 		}
 
 		# User/IP blocking
-		$block = Block::newFromTarget( $this->getName(), $ip, !$bFromSlave );
+		$block = Block::newFromTarget( $this, $ip, !$bFromSlave );
```
Comment (macbre) [inline includes/User.php:1353]: :+1:

File: tests/unit/includes/BlockTest.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	const NOT_BLOCKED_ANON = '1.1.1.1';
+
+	private $wgMemc;
+
+	protected function setUp() {
+		parent::setUp();
+
+		$this->wgMemc = $GLOBALS['wgMemc'];
+		$GLOBALS['wgMemc'] = new EmptyBagOStuff();
+	}
```
Comment (macbre) [inline tests/unit/includes/BlockTest.php:34]: We have `WikiaDatabaseTest::disableMemCache` helper for, well, disabling memcache...

**Review by macbre — APPROVED** [review]
Let's run Selenium tests suite for handling blocked users before merging this.

File: tests/unit/includes/BlockTest.php (hunk 3 above)
Comment (mszabo-wikia) [reply tests/unit/includes/BlockTest.php:34]: That's only available in `WikiaBaseTest`, and trying to make `WikiaDatabaseTest` subclass `WikiaBaseTest` breaks `WikiaMockProxy` and associated hacks 😞

**Pull request merged**